@navikt/core/react/src/form/combobox/SelectedOptions/selectedOptionsContext.tsx

```diff
@@ -101,13 +101,16 @@ const SelectedOptionsProvider = ({
     [customOptions, onToggleSelected, removeCustomOption],
   );
 
-  const isLimitReached =
-    (!!maxSelected?.limit && selectedOptions.length >= maxSelected.limit) ||
-    (!isMultiSelect && selectedOptions.length > 0);
+  const isLimitReached = useMemo(() => {
+    return (
+      (!!maxSelected?.limit && selectedOptions.length >= maxSelected.limit) ||
+      (!isMultiSelect && selectedOptions.length > 0)
+    );
+  }, [maxSelected, selectedOptions, isMultiSelect]);
 
   useEffect(() => {
     setHideCaret(isLimitReached);
-  }, [selectedOptions, setHideCaret, isLimitReached]);
+  }, [isLimitReached, setHideCaret]);
```

Comment:
Fixed 👍 

> But you can probably remove setHideCaret since it comes from useState().
Since it comes from a context and is not locally declared we (and the linter) can not guarantee that its a `DispatchEvent` from react in this case.

Reply:
Good point